src/Symfony/Bundle/FrameworkBundle/Security/SecurityHelperTrait.php

```diff
@@ -0,0 +1,167 @@
+            if (!$this->container->has('security.authorization_checker')) {
+                throw new LogicException('The SecurityBundle is not registered in your application.');
+            }
+
+            $this->authorizationChecker = $this->container->get('security.authorization_checker');
+        }
+
+        return $this->authorizationChecker->isGranted($attributes, $object);
+    }
+
+    /**
+     * Throws an exception unless the attributes are granted against the current authentication token and optionally
+     * supplied object.
+     *
+     * @param mixed  $attributes The attributes
+     * @param mixed  $object     The object
+     * @param string $message    The message passed to the exception
+     *
+     * @throws AccessDeniedException
+     */
+    protected function denyAccessUnlessGranted($attributes, $object = null, $message = 'Access Denied.')
+    {
+        if (!$this->isGranted($attributes, $object)) {
+            throw $this->createAccessDeniedException($message);
+        }
+    }
+
+    /**
+     * Returns an AccessDeniedException.
+     *
+     * This will result in a 403 response code. Usage example:
+     *
+     *     throw $this->createAccessDeniedException('Unable to access this page!');
+     *
+     * @param string          $message  A message
+     * @param \Exception|null $previous The previous exception
+     *
+     * @return AccessDeniedException
+     */
+    protected function createAccessDeniedException($message = 'Access Denied.', \Exception $previous = null)
+    {
+        return new AccessDeniedException($message, $previous);
+    }
+
+    /**
+     * Get a user from the Security Token Storage.
+     *
+     * @return mixed
+     *
+     * @see TokenInterface::getUser()
+     */
+    protected function getUser()
+    {
+        if ($this->tokenStorage === null) {
+            if (!isset($this->container)) {
+                throw new LogicException(
+                    'Unable to load the token storage. Please set the $tokenStorage property or make '
+                        .__CLASS__.' container-aware.'
+                );
+            }
+            if (!$this->container->has('security.token_storage')) {
+                throw new LogicException('The SecurityBundle is not registered in your application.');
+            }
+
+            $this->tokenStorage = $this->container->get('security.token_storage');
+        }
+
+        if (null === $token = $this->tokenStorage->getToken()) {
+            return;
+        }
+
+        if (!is_object($user = $token->getUser())) {
+            // e.g. anonymous authentication
+            return;
+        }
+
+        return $user;
+    }
+
+    /**
+     * Checks the validity of a CSRF token.
+     *
+     * @param string $id    The id used when generating the token
+     * @param string $token The actual token sent with the request that should be validated
+     *
+     * @return bool
+     */
+    protected function isCsrfTokenValid($id, $token)
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Avoid variables with short names like $id. Configured minimum length is 3.](https://www.codacy.com/app/git_9/symfony/file/2332002613/issues/source?bid=3207821&fileBranchId=3207915#l149)